### PR TITLE
Derive: serial connection of springs

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py [hunk 1]
```diff
@@ -27,7 +32,35 @@
 
 law = Eq(total_compliance, first_compliance + second_compliance)
 
-# TODO: derive law from Hooke's law
+# Derive law from Hooke's law
+## In the case of parallel connection, the force acting on both springs are equal whereas
```
Comment (blackyblack) [inline symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:36]: But it is not parallel connection.

Comment (blackyblack) [inline symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:36]:
```suggestion
## In the case of parallel connection, the forces acting on both springs are equal whereas
```

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py [hunk 2]
```diff
@@ -27,7 +32,35 @@
...
+
+_stiffness_expr = solve(compliance_def.definition, compliance_def.stiffness)[0]
+_first_stiffness = _stiffness_expr.subs(compliance_def.compliance, first_compliance)
+_second_stiffness = _stiffness_expr.subs(compliance_def.compliance, second_compliance)
+_total_stiffness = _stiffness_expr.subs(compliance_def.compliance, total_compliance)
+
+_deformation_expr = solve(
+    hookes_law.law, hookes_law.deformation
+)[0].subs(
+    hookes_law.spring_reaction, _force,
```
Comment (blackyblack) [inline symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:49]: Do we have any way to justify that force is the same on both springs?

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py (hunk 1 above)
Comment (alesanter) [reply symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:36]: I mistyped

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py (hunk 2 above)
Comment (alesanter) [reply symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:49]: Hm, yes, using 3rd Newton's law of motion, for instance. I'll add it

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py [hunk 3]
```diff
@@ -27,7 +31,48 @@
...
+# Derive law from Hooke's law
+## In the case of serial connection, the forces acting on both springs are equal whereas
+## their deformations add up.
+
+# When external force is applied to the first spring, it produces a reaction force on both sides of
+# that spring. That in turn produces a reaction force on both sides of the other spring.
+_external_force = SymSymbol("_external_force")
+_first_spring_reaction = abs(newtons_third_law.law.rhs.subs(newtons_third_law.force_action, _external_force))
+_second_spring_reaction = abs(newtons_third_law.law.rhs.subs(newtons_third_law.force_action, _first_spring_reaction))
+_total_reaction = abs(_external_force)
```
Comment (blackyblack) [inline symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:43]: Probably should also use 3rd law here: `_total_reaction = abs(newtons_third_law.law.rhs.subs(newtons_third_law.force_action, _second_spring_reaction))`

**Review by blackyblack — APPROVED** [review]

File: symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py (hunk 3 above)
Comment (alesanter) [reply symplyphysics/laws/dynamics/springs/compliance_of_two_serial_springs.py:43]: I agree

**Review by blackyblack — APPROVED** [review]

**Pull request merged**